src/Processors/Formats/IInputFormat.h

```diff
@@ -45,6 +47,26 @@ struct ChunkInfoRowNumbers : public ChunkInfo
     std::optional<IColumnFilter> applied_filter;
 };
 
+struct FileBucketInfo
+{
+    virtual void serialize(WriteBuffer & buffer) = 0;
+    virtual void deserialize(ReadBuffer & buffer) = 0;
+    virtual String getIdentifier() const = 0;
+    virtual std::shared_ptr<FileBucketInfo> createFromBuckets(std::vector<size_t> buckets_ids) = 0;
+    virtual std::shared_ptr<FileBucketInfo> clone() const = 0;
+    virtual String getFormatName() const = 0;
+
+    virtual ~FileBucketInfo() = default;
+};
+using FileBucketInfoPtr = std::shared_ptr<FileBucketInfo>;
+
+struct IBucketSplitter
```

Comment:
The same here, please, write a description for this structure and what `splitToBuckets` method should do